src/components/NeBadge.vue

```diff
@@ -0,0 +1,147 @@
+<script lang="ts" setup>
+import { computed, PropType } from 'vue'
+import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
+
+const props = defineProps({
+  /**
+   * Size of the badge, can be 'sm' or 'xs', defaults to 'sm'.
+   */
+  size: {
+    type: String as PropType<'sm' | 'xs'>,
+    default: 'sm'
+  },
+  /**
+   * Badge type, defaults to 'secondary'
+   */
+  theme: {
+    type: String as PropType<
+      'primary' | 'secondary' | 'tertiary' | 'error' | 'warning' | 'success' | 'info'
+    >,
+    default: 'secondary'
+  },
+  /**
+   * Should the component be rounded or not.
+   */
+  rounded: {
+    type: Boolean,
+    default: true
+  },
+  /**
+   * Text to display inside the badge.
+   */
+  text: {
+    type: String,
+    required: true
+  },
+  /**
+   * Label below the text.
+   */
+  label: {
+    type: String,
+    required: false
+  },
+  /**
+   * Icon to display inside the badge.
+   */
+  icon: {
+    type: Array<string>,
+    required: false
+  },
+  /**
+   * Position of the icon inside the badge.
+   */
+  iconPosition: {
+    type: String as PropType<'left' | 'right'>,
+    default: 'left'
+  }
+})
+
+/**
+ * Returns the styling classes based off the size prop.
+ */
+const textClasses = computed(() => {
+  switch (props.size) {
+    case 'xs':
+      return 'text-xs'
+    default:
+      return 'text-sm'
+  }
+})
+
+/**
+ * Returns padding classes base off size and presence of label.
+ */
+const paddingClasses = computed(() => {
+  if (props.label) {
+    // if icon is present, check where it is to decrease padding on that side
+    if (props.icon) {
+      if (props.iconPosition == 'left') {
+        return 'pl-3 pr-4'
+      } else {
+        return 'pl-4 pr-3'
+      }
+    } else {
+      return 'px-4'
+    }
+  } else {
+    return 'px-2.5'
+  }
+})
+
+/**
+ * Spacing classes separating
+ */
+const spacingClasses = computed(() => {
+  if (props.label) {
+    return 'gap-x-2'
+  } else {
+    return 'gap-x-1'
+  }
+})
+
+/**
+ * Returns the styling classes based off the type prop.
+ */
+const typeClasses = computed(() => {
+  switch (props.theme) {
+    case 'primary':
+      return 'bg-primary-100 text-primary-800 dark:bg-primary-700 dark:text-primary-100'
+    case 'tertiary':
+      return 'bg-indigo-100 text-indigo-800 dark:bg-indigo-700 dark:text-indigo-100'
+    case 'success':
+      return 'bg-green-100 text-green-800 dark:bg-green-700 dark:text-green-50'
+    case 'warning':
+      return 'bg-amber-100 text-amber-800 dark:bg-amber-700 dark:text-amber-50'
+    case 'error':
+      return 'bg-rose-100 text-rose-800 dark:bg-rose-700 dark:text-rose-100'
+    case 'info':
+      return 'bg-blue-100 text-blue-800 dark:bg-blue-700 dark:text-blue-100'
+    default:
+      return 'bg-gray-100 text-gray-800 dark:bg-gray-600 dark:text-gray-100'
```

Comment:
```suggestion
      return 'bg-gray-200 text-gray-800 dark:bg-gray-600 dark:text-gray-100'
```
`bg-gray-100` is sometimes used as background color for pages and other containers; if we use it for badge background it might be invisible (for "secondary" badges this happens in the storybook too)